### PR TITLE
New Header-V2

### PR DESCRIPTION
**Pull request by cp-priyanka-g**

### CODE REVIEW

**Comment by cp-shruti-s** [comment]
Screencast from 07-04-22 03_26_54 PM IST

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+              <a class="nav-link" :href="careerURL">Career</a>
+            </li>
+            <li class="nav-item-margin v2-title-3-text">
+              <a class="nav-link" target="_blank">Blogs</a>
+            </li>
+            <li class="nav-item-margin v2-title-3-text">
+              <a class="nav-link" target="_blank">Portfolio</a>
+            </li>
+            <li class="nav-item-margin round-btn v2-title-3-text">
+              <a class="anchor-color" target="_blank">Let's Talk</a>
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:38]: Add hover effect in Let's Talk button 

-     Button background color white and text black

File: vue-frontend/package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "@babel/eslint-parser": "^7.16.5",
     "@vue/cli-plugin-babel": "~5.0.0-rc.2",
     "@vue/cli-plugin-eslint": "~5.0.0-rc.2",
-    "@vue/cli-service": "^5.0.1",
+    "@vue/cli-service": "^5.0.4",
```
Comment (cp-shruti-s) [inline vue-frontend/package.json:35]: Merge your MR with master

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            </li>
+            <li class="nav-item-margin v2-title-3-text">
+              <a class="nav-link" :href="blogsURL" target="_blank">Blogs</a>
+            </li>
+            <li class="nav-item-margin v2-title-3-text">
+              <a class="nav-link" target="_blank">Portfolio</a>
+            </li>
+            <router-link :to="contactURL">
+              <li class="nav-item-margin round-btn v2-title-3-text">
+                <a class="anchor" target="_blank">Let's Talk</a>
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:39]: - I think you make a link instead for let's talk button in mobile view 
- We all ready implement in our exciting website

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+a {
+  color: #000;
+}
+
+.navbar-animation {
+  animation: menu_sticky 0.6s ease-in-out;
+}
+
+.bottom-border::after {
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:134]: Don't need to write `bottom border` css over here because ewe already define `underline-text` css class in `App.vue` class

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+.navbar-nav {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: flex-start;
+}
+
+.navbar-nav .nav-item-margin .nav-link {
+  margin: 0 20px 0 0;
+  color: rgb(0, 0, 0);
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:163]: And which color is it??

Comment (cp-priyanka-g) [reply vue-frontend/src/components/partials/HeaderV2.vue:163]: black color

File: vue-frontend/src/components/partials/HeaderV2.vue (hunk 4 above)
Comment (cp-priyanka-g) [reply vue-frontend/src/components/partials/HeaderV2.vue:134]: okay i will remove it

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    color: black;
+  }
+  .round-btn:hover {
+    background-color: white;
+    color: black;
+  }
+  .anchor {
+    color: white;
+  }
+  .col-md {
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:202]: Why bootstrap class define over here

Comment (cp-priyanka-g) [reply vue-frontend/src/components/partials/HeaderV2.vue:202]: no its a userdefined. should i have to change the naming convention ?

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+  .navbar {
+    padding: 20px 0;
+  }
+  .col-md {
+    flex-direction: column;
+  }
+}
+
+@include media-breakpoint-up(lg) {
+  .round-btn {
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:185]: Define this button css in App.vue file because we want the code to be common

File: vue-frontend/src/components/partials/HeaderV2.vue (hunk 6 above)
Comment (cp-shruti-s) [reply vue-frontend/src/components/partials/HeaderV2.vue:202]: No, it totally removes and refers to the `underline-text` css class that we used in the All Ready Exciting website.

File: vue-frontend/src/components/partials/HeaderV2.vue (hunk 5 above)
Comment (cp-shruti-s) [reply vue-frontend/src/components/partials/HeaderV2.vue:163]: In this color define navigation link in Fing

File: vue-frontend/src/App.vue [hunk 8]
```diff
@@ -233,6 +233,21 @@ a {
...
 }
+.round-btn {
+  background-color: #000;
+  border-radius: 25px;
+  padding: 10px 25px;
+  color: #fff;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
+  text-decoration: none;
+}
+.round-btn:hover > a {
```
Comment (cp-shruti-s) [inline vue-frontend/src/App.vue:244]: Add hover effect in side `@media` query 
`@media (hover: hover) and (pointer: fine) {}`

Comment (cp-shruti-s) [inline vue-frontend/src/App.vue:244]: Add hover effect in side `@media` query 
`@media (hover: hover) and (pointer: fine) {}`

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+    justify-content: flex-start;
+    align-items: flex-start;
+  }
+}
+
+@include media-breakpoint-up(lg) {
+  .col-md {
+    flex-direction: row;
+  }
+  .round-btn {
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:198]: This css also apply in `App.vue` file

File: vue-frontend/src/pages/index.vue [hunk 10]
```diff
@@ -14,6 +14,7 @@ export default {
   data() {
     return {
       isShowNewHomePage: config.IS_SHOW_NEW_HOME_PAGE,
+      isNewHeader: config.IS_NEW_HEADER,
```
Comment (cp-shruti-s) [inline vue-frontend/src/pages/index.vue:17]: In this page, no need feature flag over here

File: vue-frontend/src/config.js [hunk 11]
```diff
@@ -34,4 +34,5 @@ export default Object.freeze({
   GAP_ID_ROOT: "",
   IS_PROD: false,
   IS_SHOW_NEW_HOME_PAGE: true,
+  IS_NEW_HEADER: true,
```
Comment (cp-shruti-s) [inline vue-frontend/src/config.js:37]: Also add this feature flag in` config.prod.js` file

File: vue-frontend/src/App.vue [hunk 12]
```diff
@@ -176,7 +176,7 @@ a {
   font-size: 1rem;
   line-height: 1.25rem;
   letter-spacing: 1px;
-  color: #3d3d3d;
+  color: #3d3d3d !important;
```
Comment (cp-sumi-k) [inline vue-frontend/src/App.vue:179]: We should not use `important` if possible

File: vue-frontend/src/App.vue [hunk 13]
```diff
@@ -240,19 +245,39 @@ a {
 .v2-button > span {
   margin-right: 6px;
 }
+.v2-black-btn {
```
Comment (cp-sumi-k) [inline vue-frontend/src/App.vue:248]: We can reuse portfolio buttons here.

File: vue-frontend/src/App.vue [hunk 14]
```diff
@@ -240,19 +245,39 @@ a {
...
+  background-color: #3d3d3d;
+  color: #fff !important;
+  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.5);
+}
 
 @media (hover: hover) and (pointer: fine) {
   .v2-button:hover {
     background-color: #3d3d3d;
-    color: #fff;
+    color: #fff !important;
```
Comment (cp-sumi-k) [inline vue-frontend/src/App.vue:260]: Again should not use `important`

File: vue-frontend/src/components/home-new/LandingSection.vue [hunk 15]
```diff
@@ -0,0 +1,115 @@
+<template>
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/home-new/LandingSection.vue:1]: This changes should not be in this MR as its separate MR and not merged yet

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+                href="/"
+                :style="[
+                  activeHomePath
+                    ? {
+                        textDecoration: `underline !important`,
+                        'text-underline-offset': '0.3rem',
+                      }
+                    : { textDecoration: `none` },
+                ]"
+                ><span :class="[activeHomePath ? '' : 'v2-underline-text']"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/partials/HeaderV2.vue:35]: We can add all links in an array and display them here in for loop instead of doing them all separately, also add an active class on click of a link.

File: vue-frontend/src/pages/index.vue [hunk 17]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div>
-    <HomeIndex v-if="!isShowNewHomePage" />
+    <HomeIndex v-if="isShowNewHomePage" />
```
Comment (cp-shruti-s) [inline vue-frontend/src/pages/index.vue:3]: Do not change this file

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
+      activeHomePath: false,
+      activeCareerPath: false,
+      activeContactPath: false,
+    };
+  },
+  components: {},
+  mounted() {
+    let currentRoutePath = this.$router.currentRoute._value.path;
+
+    this.activeHomePath = currentRoutePath == "/";
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/partials/HeaderV2.vue:103]: We can remove all separate variables if its taken in array and handle `active` on click

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+}
+@include media-breakpoint-up(lg) {
+  .col-flow {
+    flex-direction: row;
+  }
+  .v2-black-btn {
+    display: block;
+    padding: 10px 30px;
+  }
+  .btn-link {
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/partials/HeaderV2.vue:227]: This class was also added in app.vue, what is the purpose of that?

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+              <a class="nav-link" :href="blogsURL" target="_blank">Blogs</a>
+            </li>
+            <li class="nav-item-margin v2-title-3-text">
+              <a class="nav-link" target="_blank">Portfolio</a>
+            </li>
+            <li>
+              <a
+                :href="contactURL"
+                class="nav-item-margin v2-title-3-text"
+                v-bind:class="activeContactPath ? 'v2-button' : 'v2-black-btn'"
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:53]: Add button border at hover time

File: vue-frontend/src/App.vue (hunk 12 above)
Comment (cp-priyanka-g) [reply vue-frontend/src/App.vue:179]: i have used .v2-title-3-text  in which you used important in font color hence this overriding my button font color so I need to use !important. Can I remove that !important in  .v2-title-3-text   in  app.vue

File: vue-frontend/src/components/partials/HeaderV2.vue (hunk 19 above)
Comment (cp-priyanka-g) [reply vue-frontend/src/components/partials/HeaderV2.vue:227]: it create link of black button when view in mobile or tablet view

File: vue-frontend/src/components/partials/HeaderV2.vue [hunk 21]
```diff
@@ -0,0 +1,232 @@
...
+        </router-link>
+        <div class="navbar-collapse">
+          <ul class="navbar-nav ms-auto mb-2 mb-lg-0">
+            <li class="nav-item-margin list-flow">
+              <a
+                class="nav-link v2-title-3-text"
+                v-for="item in items"
+                :key="item.id"
+                :style="[
+                  currentRoutePath == item.url
```
Comment (cp-shruti-s) [inline vue-frontend/src/components/partials/HeaderV2.vue:29]: Can you please explain me this condition ??
